packages/shared/src/components/checklist/SquadFirstCommentChecklistStep.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { ReactElement } from 'react';
+import { useRouter } from 'next/router';
+import { ChecklistStepProps } from '../../lib/checklist';
+import { Button } from '../buttons/Button';
+import { ChecklistStep } from './ChecklistStep';
+import { Squad } from '../../graphql/sources';
+import { useFindSquadWelcomePost } from '../../hooks/useFindSquadWelcomePost';
+
+const SquadFirstCommentChecklistStep = ({
+  squad,
+  ...props
+}: ChecklistStepProps & { squad: Squad }): ReactElement => {
+  const router = useRouter();
+  const welcomePost = useFindSquadWelcomePost(squad);
+
+  return (
+    <ChecklistStep {...props}>
+      <Button
+        className="btn-primary"
+        disabled={!welcomePost}
+        onClick={() => {
+          // TODO WT-1313-squad-welcome-post-checklist-steps we might need to open article modal here
+          router.push(`/posts/${welcomePost.id}?comment`);
```

Comment:
Lee actually has the same caveat now from BE perspective my suggestion was:

`posts/{id}?comment={plaintext}`

So maybe align on who will tackle which part :)
Great to have a generic solution like that.